src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -0,0 +1,1567 @@
+      Arrays.fill(this.qFlag, 0, this.numYears, 0);
+      Arrays.fill(this.cstSeg, 0, this.numYears, 0);
+      Arrays.fill(this.cstSegSmooth, 0, this.numYears, 0);
+      Arrays.fill(this.distFlag, 0, this.numYears, 0);
+      Arrays.fill(this.distYear, 0, this.numYears, 0);
+      Arrays.fill(this.distLength, 0, this.numYears, 0);
+      Arrays.fill(this.regrFlag, 0, this.numYears, 0);
+
+      Arrays.fill(this.distR2, 0, this.numYears, 0.0);
+      Arrays.fill(this.distMagn, 0, this.numYears, 0.0);
+      Arrays.fill(this.distMagnB4, 0, this.numYears, 0.0);
+      Arrays.fill(this.distMagnVi, 0, this.numYears, 0.0);
+      Arrays.fill(this.distMagnBr, 0, this.numYears, 0.0);
+      Arrays.fill(this.regrR2, 0, this.numYears, 0.0);
+      Arrays.fill(this.regrSlope, 0, this.numYears, 0.0);
+      Arrays.fill(this.regrRough, 0, this.numYears, 0.0);
+
+      Arrays.fill(this.meanForUdBx, 0, N_BANDS, 0.0);
+      Arrays.fill(this.sdForUdBx, 0, N_BANDS, 0.0);
+    }
+
+    /**
+     * Classify the pixel's mask values into QA_GOOD and QA_BAD and fill in
+     * QA_BAD values based on nearest neighbors or linear interpolation of
+     * QA_GOOD values. Implements section 3.3.1 in Huang et al. (2010) paper
+     */
+    private void interpolationAndIndices() {
+
+      // Flag each year's pixel as good or bad based on the mask value
+      // TODO: Note that class 0 (BACKGROUND) is *not* being flagged as 
+      // "fillable" in the original source code (ie. SLC-off errors).  
+      // We need to ask Cheng about this
+      for (int i = 0; i < this.numYears; i++) {
+        if (this.mask[i] != 0 && this.mask[i] <= FILL_CLASSES) {
+          this.qFlag[i] = QA_BAD;
+        } else {
+          this.qFlag[i] = QA_GOOD;
+        }
+      }
+
+      // Look for spikes and dips that may be unflagged cloud and shadow and
+      // set the QA flag accordingly
+      for (int i = 1; i < this.numYears - 1; i++) {
+
+        // Skip over pixels already labeled as QA_BAD
+        if (this.qFlag[i] == QA_BAD) {
+          continue;
+        }
+
+        // Pixels higher in UD than neighboring years - cloud
+        if (isRelativeCloud(i)) {
+          this.qFlag[i] = QA_BAD;
+          continue;
+        }
+
+        // Pixels lower than UD than neighboring years - shadow
+        if (isRelativeShadow(i)) {
+          this.qFlag[i] = QA_BAD;
+        }
+      }
+
+      // Now test the start/end conditions
+      int n = this.numYears;
+      if (this.qFlag[0] == QA_GOOD && isBadEndpoint(0, 1)) {
+        this.qFlag[0] = QA_BAD;
+      }
+      if (this.qFlag[n - 1] == QA_GOOD && isBadEndpoint(n - 1, n - 2)) {
+        this.qFlag[n - 1] = QA_BAD;
+      }
+
+      // Calculate bad percentage
+      int badCount = 0;
+      for (int i = 0; i < this.numYears; i++) {
+        if (this.qFlag[i] == QA_BAD) {
+          badCount += 1;
+        }
+      }
+      this.pctGoodObs = (int) (100.0 - ((100.0 * badCount) / this.numYears));
```

Comment:
Yep, good catch.  You will find there are a number of global (class) variables that aren't being used (e.g. vRange, hudSeg, ludSeg, etc.).  In original VCT, these were being output as rasters.  I've left them in for now just so that we have a record of the full port, but it's probably good to get rid of anything that isn't expressly needed for the later ensemble modeling (or what a typical VCT user might want ...)